registration/registration-services/src/main/java/io/mosip/registration/update/SoftwareUpdateHandler.java

```diff
@@ -548,96 +551,97 @@ public Timestamp getLatestVersionReleaseTimestamp() {
+		}
+		
+		for (Entry<String, VersionMappings> entry : versionMappings.entrySet()) {
+			try {
+				executeSQL(entry.getValue().getDbVersion());
+				previousVersion = entry.getKey();
+				// Update global param with current version
+				globalParamService.update(RegistrationConstants.SERVICES_VERSION_KEY, entry.getKey());
+			} catch (RegBaseCheckedException exception) {
+				// Prepare Error Response
+				setErrorResponse(responseDTO, RegistrationConstants.SQL_EXECUTION_FAILURE, null);
+				// Replace with backup
+				rollback(responseDTO, previousVersion);
+				return responseDTO;
+			}
+		}
+		
+		setSuccessResponse(responseDTO, RegistrationConstants.SQL_EXECUTION_SUCCESS, null);
+
+		LOGGER.info(LoggerConstants.LOG_REG_UPDATE, APPLICATION_NAME, APPLICATION_ID,
+				"DB-Script files execution completed");
 
-		// execute sql file
+		return responseDTO;
+	}
 
+	private void executeSQL(String dbVersion) throws RegBaseCheckedException {
 		try {
-
 			LOGGER.info(LoggerConstants.LOG_REG_UPDATE, APPLICATION_NAME, APPLICATION_ID,
-					"Checking Started : " + newVersion + SLASH + exectionSqlFile);
+					"Checking Started : " + dbVersion + SLASH + exectionSqlFile);
 
-			execute(SQL + SLASH + newVersion + SLASH + exectionSqlFile);
+			execute(SQL + SLASH + dbVersion + SLASH + exectionSqlFile);
 
 			LOGGER.info(LoggerConstants.LOG_REG_UPDATE, APPLICATION_NAME, APPLICATION_ID,
-					"Checking completed : " + newVersion + SLASH + exectionSqlFile);
-
-		}
-
-		catch (RuntimeException | IOException runtimeException) {
-
+					"Checking completed : " + dbVersion + SLASH + exectionSqlFile);
+		} catch (RuntimeException | IOException runtimeException) {		
 			LOGGER.error(LoggerConstants.LOG_REG_UPDATE, APPLICATION_NAME, APPLICATION_ID,
 					runtimeException.getMessage() + ExceptionUtils.getStackTrace(runtimeException));
 
 			// ROLL BACK QUERIES
 			try {
-
 				LOGGER.info(LoggerConstants.LOG_REG_UPDATE, APPLICATION_NAME, APPLICATION_ID,
-						"Checking started : " + newVersion + SLASH + rollBackSqlFile);
+						"Checking started : " + dbVersion + SLASH + rollBackSqlFile);
 
-				execute(SQL + SLASH + newVersion + SLASH + rollBackSqlFile);
+				execute(SQL + SLASH + dbVersion + SLASH + rollBackSqlFile);
 
 				LOGGER.info(LoggerConstants.LOG_REG_UPDATE, APPLICATION_NAME, APPLICATION_ID,
-						"Checking completed : " + newVersion + SLASH + rollBackSqlFile);
-
+						"Checking completed : " + dbVersion + SLASH + rollBackSqlFile);
 			} catch (RuntimeException | IOException exception) {
-
 				LOGGER.error(LoggerConstants.LOG_REG_UPDATE, APPLICATION_NAME, APPLICATION_ID,
 						exception.getMessage() + ExceptionUtils.getStackTrace(exception));
-
 			}
-			// Prepare Error Response
-			setErrorResponse(responseDTO, RegistrationConstants.SQL_EXECUTION_FAILURE, null);
-
-			// Replace with backup
-			rollback(responseDTO, previousVersion);
```

Comment:
Are we handling the backup restore in someother place?

Reply:
No, we only have it at this place.